endpoints/openrtb2/amp_auction.go

```diff
@@ -73,7 +73,7 @@ func NewAmpEndpoint(
 	tmaxAdjustments *exchange.TmaxAdjustmentsPreprocessed,
 ) (httprouter.Handle, error) {
 
-	if ex == nil || validator == nil || requestsById == nil || accounts == nil || cfg == nil || metricsEngine == nil {
```

Comment:
Why do we no longer need this nill check?

Reply:
Ah that was removed when I had the validator as a struct. I re-added the nil checks now that it is an interface.